internal/controllers/machineset/machineset_preflight_test.go

```diff
@@ -0,0 +1,472 @@
+				machineSet: &clusterv1.MachineSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+					},
+					Spec: clusterv1.MachineSetSpec{
+						Template: clusterv1.MachineTemplateSpec{
+							Spec: clusterv1.MachineSpec{
+								Version: pointer.String("v1.27.0"),
+							},
+						},
+					},
+				},
+				wantPass: false,
+			},
+			{
+				name: "kubernetes version preflight check: should fail if the machine set minor version is 2 older than control plane minor version",
+				cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+					},
+					Spec: clusterv1.ClusterSpec{
+						ControlPlaneRef: contract.ObjToRef(controlPlaneStable),
+					},
+				},
+				controlPlane: controlPlaneStable,
+				machineSet: &clusterv1.MachineSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+					},
+					Spec: clusterv1.MachineSetSpec{
+						Template: clusterv1.MachineTemplateSpec{
+							Spec: clusterv1.MachineSpec{
+								Version: pointer.String("v1.23.0"),
+							},
+						},
+					},
+				},
+				wantPass: false,
+			},
+			{
+				name: "kubernetes version preflight check: should pass if the machine set minor version is greater than control plane minor version but the preflight check is skipped",
+				cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+					},
+					Spec: clusterv1.ClusterSpec{
+						ControlPlaneRef: contract.ObjToRef(controlPlaneStable),
+					},
+				},
+				controlPlane: controlPlaneStable,
+				machineSet: &clusterv1.MachineSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+						Annotations: map[string]string{
+							clusterv1.MachineSetSkipPreflightChecksAnnotation: string(clusterv1.MachineSetPreflightCheckKubernetesVersion),
+						},
+					},
+					Spec: clusterv1.MachineSetSpec{
+						Template: clusterv1.MachineTemplateSpec{
+							Spec: clusterv1.MachineSpec{
+								Version: pointer.String("v1.27.0"),
+							},
+						},
+					},
+				},
+				wantPass: true,
+			},
+			{
+				name: "kubernetes version preflight check: should pass if the machine set minor version and control plane version conform to kubernetes version skew policy",
+				cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+					},
+					Spec: clusterv1.ClusterSpec{
+						ControlPlaneRef: contract.ObjToRef(controlPlaneStable),
+					},
+				},
+				controlPlane: controlPlaneStable,
+				machineSet: &clusterv1.MachineSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ns,
+					},
+					Spec: clusterv1.MachineSetSpec{
+						Template: clusterv1.MachineTemplateSpec{
+							Spec: clusterv1.MachineSpec{
+								Version: pointer.String("v1.25.6"),
+							},
+						},
+					},
+				},
+				wantPass: true,
+			},
+			{
+				name: "kubeadm version preflight check: should fail if the machine set version is not equal to control plane version when using kubeadm bootstrap provider",
```

Comment:
As far as I can tell major + minor are the same, this should pass?

Reply:
Note: This only fails because it fails the Kubernetes version skew preflight check